file-uploader/inject/injector.go

```diff
@@ -0,0 +1,52 @@
+//+build wireinject
+
+package inject
+
+import (
+	"github.com/ONSdigital/ras-rm-sample/file-uploader/config"
+	"github.com/ONSdigital/ras-rm-sample/file-uploader/file"
+	"github.com/google/wire"
+	"github.com/spf13/viper"
+	"cloud.google.com/go/pubsub"
+	"context"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var FileProcessor = Inject()
+
+func Inject() file.FileProcessor{
+	wire.Build(NewFileProcessor, ConfigSetup, GenContext, NewPubSub)
```

Comment:
is this the dependency injection thing?

Reply:
yes this injector is a result of me thinking in java terms. the default way of dependency injection is not something im particularly keen on. Wire offers compile time dependency injection and looks interesting but I must admit its probably overkill for this

Reply:
I don't see any reason not to use it. Might have a look into it myself.

Might have to knock that Java out of you though ;)